src/ServiceInsight/Framework/Commands/CloseWindowCommand.cs

```diff
@@ -0,0 +1,32 @@
+﻿namespace ServiceInsight.Framework.Commands
+{
+    using System;
+    using System.Windows;
+    using System.Windows.Input;
+
+    public class CloseWindowCommand : ICommand
+    {
+        readonly Window target;
+
+        public CloseWindowCommand(Window target)
+        {
+            this.target = target;
+        }
+
+        public bool CanExecute(object parameter)
+        {
+            return target != null;
+        }
+
+        public void Execute(object parameter)
+        {
+            if (CanExecute(parameter))
+            {
+                target.Close();
+            }
+        }
+
+        #pragma warning disable CS0067
+        public event EventHandler CanExecuteChanged;
```

Comment:
We should really re-enable the warning again.

```CSharp
#pragma warning disable CS0067
public event EventHandler CanExecuteChanged;
#pragma warning restore CS0067

Reply:
Thanks @WilliamBZA forgot about that :)

Reply:
I saw that too but because it was at the end of the class I let it go. However, 👍 for correctness. I'm ashamed of myself. 😜